lib/AST/IRToASTVisitor.cpp

```diff
@@ -398,8 +398,28 @@ void IRToASTVisitor::VisitArgument(llvm::Argument &arg) {
   // Get parent function declaration
   auto func{arg.getParent()};
   auto fdecl{clang::cast<clang::FunctionDecl>(GetOrCreateDecl(func))};
+  auto argtype{arg.getType()};
+  if (arg.hasByValAttr()) {
+    argtype = llvm::cast<llvm::PointerType>(argtype)->getElementType();
```

Comment:
Is the pointer element type communicated via the `byval` attribute when opaque pointer types are used?

Reply:
I'm not sure what you mean by 'opaque pointer types'
As the call is by value, the type must be complete, I'm not sure how the pointer type could become opaque

Reply:
I think newer LLVM versions support pointer types without an element type. Here's a concrete `byval`-related example: https://llvm.org/docs/OpaquePointers.html#i-still-need-pointee-types

Reply:
Addressed in latest commit